### ISSUE TITLE
Controlling styling / modifications to D3FC charts

### ISSUE THREAD

**Issue by limx0** [finos/perspective#1487]
I'm currently streaming data from the tornado backend to a couple of perspective widgets in a html file (thank you for the plentiful examples!) one of which is a rolling time series scatter plot with multiple series. Currently, I believe, the mapping of colours/markers in the `X/Y scatter` plot is basically just the enumeration of whatever data exists in the table at a given time. This is causing my series to change colour/marker if one of the columns in the table is all null for some period of time. Is it possible to declare these styling attributes via some other config?

I wonder if it is within scope for perspective to allow configuring the plots from table values (for example like holoviews does - https://holoviews.org/user_guide/Style_Mapping.html)?

**Comment by limx0** [comment]
Hey @texodus - sorry for the tag but I see you've been hard at work on some UI improvements (which look awesome!) so I just wanted to bump and see if anything like this was on the radar (specifically more control over D3FC via table values ala holoviews Style Mapping link above)? 

I could potentially offer a bounty for this work if that would help this along?

**Comment by Martinlopez-hash** [comment]
Yes it will help you and I a lot so do what you have to do on UI An AI
improved

**Comment by texodus** [comment]
@limx0 there three conflated issues present we'd like to address:
1) Color range of `"split_by"` (formerly `"column_pivots"`) shouldn't re-assign when the table is updated. This is a bug.
2) `"X/Y Scatter"` _also has_ a color field (column position 3), so you could use `"label"` column here, but it does not currently work with string columns. However, `"Treemap"` and `"Sunburst"` do work correctly with string columns in the color field, so this is another easy bug.
3) We're working on a few UX features to specify the styling properties of charts, but this work is still in the design stage.

I've opened issues #2198 (1) and #2199 (2) to track these bugs, which should at least fix your issue for now, so I'm going to convert this issue to a discussion, in favor of those for tracking.
